src/lib/middleware/factory.ts

```
@@ -0,0 +1,53 @@
+import { NextRequest, NextResponse } from 'next/server';
+
+// Define a more flexible Middleware type that can be chained or used standalone.
+// It can return void (if it only modifies the request/response or does not terminate),
+// or a NextResponse (if it redirects, rewrites, or ends the response).
+export type MiddlewareFunction = (
+  req: NextRequest,
+  res: NextResponse // Pass response to allow modification or use as base for new response
+) => Promise<NextResponse | void> | NextResponse | void;
```

Biome (1.9.4)

[error] void is confusing inside a union type.

Unsafe fix: Use undefined instead.


(lint/suspicious/noConfusingVoidType)